### PR TITLE
WIP - Refactor of react and react-material-ui

### LINKED ISSUE #9: Store user in state instead of using forceUpdate
Rewrite the update code since rather than calling forceUpdate the user can be just tracked in the component state.

### LINKED ISSUE #11: Improve styling of error messages in material-ui package
In the current implementation the errors which surface from server responses are displayed at the bottom of the login and signup forms. This text needs to be styled to match the rest of the material ui theme. It should probably derive its color from the mui context to work with custom themes easily.

### LINKED ISSUE #14: Allow custom logo
The logo could either supplement or replace the title. 

Consider if logo should be passed down through the config or alternatively write documentation on how to extend the Login / Signup components with custom elements.

### LINKED ISSUE #16: Don't flash login form / add loading indicator
Since session resuming is asynchronous the login form component flashes on the screen for a brief moment while the session is being resumed.

### LINKED ISSUE #17: Fix storybook for material ui package

### LINKED ISSUE #22: [withUser] doesn't pass the user object if not logged in
If you use withUser when you are not connected you receive a user object props empty, should be null instead.

### PR DESCRIPTION
**Pull request by TimMikeladze**
Closes #22 
Closes #17 
Closes #16 
Closes #14 
Closes #11 
Closes #9

### CODE REVIEW

**Comment by TimMikeladze** [comment]
@grantga

File: packages/material-ui/src/Accounts.js [hunk 1]
```diff
@@ -1,36 +1,24 @@
...
-  <Flexbox flexDirection="column" justifyContent="center" alignItems="center">
-    <div style={styles.base}>
-      <Paper style={styles.content}>
-        <Title title={AccountsClient.options().title} />
-        <AccountsBase formType={formType} />
-      </Paper>
-    </div>
-  </Flexbox>
-  );
+import { merge } from 'lodash';
```
Comment (pradel) [inline packages/material-ui/src/Accounts.js]: your import should be `import merge from 'lodash/merge';`

File: packages/material-ui/src/Content.js [hunk 2]
```diff
@@ -0,0 +1,25 @@
+import React, { PropTypes } from 'react';
+import Flexbox from 'flexbox-react';
+import {
```
Comment (pradel) [inline packages/material-ui/src/Content.js]: `import Paper from 'material-ui/Paper`

File: packages/material-ui/src/loginComponents.js [hunk 3]
```diff
@@ -0,0 +1,132 @@
+/* eslint-disable no-shadow */
+import React, { PropTypes } from 'react';
+import Flexbox from 'flexbox-react';
+import { getContext } from 'recompose';
```
Comment (pradel) [inline packages/material-ui/src/loginComponents.js]: `import getContext from 'recompose/getContext';`

**Comment by pradel** [comment]
All your imports related to lodash, material-ui and recompose should be per component and not a global require to avoid importing the all lib.

**Comment by pradel** [comment]
I think the authenticate component should pass the user prop to the children, this will avoid the use of one more withCurrentUser.

File: packages/react/src/authenticate.js [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+      } catch (err) {
+        this.props.showDialog(true);
+      }
+    },
+    componentWillUnmount() {
+      this.unsubscribe();
+    },
+  }),
+)(({ user, dialog }) => {
+  if (accounts.isLoading()) {
```
Comment (pradel) [inline packages/react/src/authenticate.js]: actually when accounts.isLoading is false and user is empty render return null, instead it should redirect to login

**Comment by pradel** [comment]
@TimMikeladze I was thinking maybe all the texts for the fields contained in material-ui like `Recover my password` etc .. should be passed as props by the react package. We will be able to have a message dictionary for the react package which is independent of the ui library. Also we will be able to reuse components. Like password field can be used in login and reset-password view since it's the only things that will change between the two

**Comment by TimMikeladze** [comment]
@pradel that's a very good point. I will make the necessary adjustments.

**Pull request merged**